doc/user/security.md

```diff
@@ -0,0 +1,119 @@
+CVE-2012-4522 | Unintentional file creation caused by inserting a illegal NUL character | *todo* | *todo* | *todo*
+CVE-2012-4464, CVE-2012-4466  | $SAFE escaping vulnerability about Exception#to_s / NameError#to_s | *todo* | *todo* | *todo*
+None | Security Fix for RubyGems: SSL server verification failure for remote repository | *todo* | *todo* | *todo*
+CVE-2011-3389 | Security Fix for Ruby OpenSSL module: Allow 0/n splitting as a prevention for the TLS BEAST attack | *todo* | *todo* | *todo*
+CVE-2011-4815 | Denial of service attack was found for Ruby's Hash algorithm (cross-reference CVE-2011-4838, CVE-2012-5370, CVE-2012-5372) | Hashes are made non-deterministic by incorporating process start time | Tested in ruby/spec `:security` | Hashes are made non-deterministic by incorporating output from `/dev/random`
+None | Exception methods can bypass $SAFE | *todo* | *todo* | *todo*
+None | FileUtils is vulnerable to symlink race attacks | *todo* | *todo* | *todo*
+CVE-2010-0541 | XSS in WEBrick | *todo* | *todo* | *todo*
+None | Buffer over-run in ARGF.inplace_mode= | *todo* | *todo* | *todo*
+None | WEBrick has an Escape Sequence Injection vulnerability | *todo* | *todo* | *todo*
+CVE-2009-4124 | Heap overflow in String | *todo* | *todo* | *todo*
+None | DoS vulnerability in BigDecimal | *todo* | *todo* | *todo*
+None | DoS vulnerability in REXML | *todo* | *todo* | *todo*
+CVE-2008-1447 | Multiple vulnerabilities in Ruby | *todo* | *todo* | *todo*
+CVE-2008-2662, CVE-2008-2663, CVE-2008-2725, CVE-2008-2726, CVE-2008-2664, CVE-2008-1891 | Arbitrary code execution vulnerabilities | *todo* | *todo* | *todo*
+None | File access vulnerability of WEBrick | *todo* | *todo* | *todo*
+None | Net::HTTPS Vulnerability | *todo* | *todo* | *todo*
+JVN#84798830 | Another DoS Vulnerability in CGI Library | *todo* | *todo* | *todo*
+CVE-2006-5467 | DoS Vulnerability in CGI Library | *todo* | *todo* | *todo*
+VU#160012 | Ruby vulnerability in the safe level settings | *todo* | *todo* | *todo*
+
+## JRuby Vulnerabilities
+
+TruffleRuby uses code from JRuby, so vulnerabilities reported against JRuby may
+apply to TruffleRuby.
+
+Number | Description | Their Mitigation | Test | Our Mitigation
+--- | --- | --- | --- | ---
+CVE-2012-5370 | JRuby computes hash values without properly restricting the ability to trigger hash collisions predictably (cross-reference CVE-2011-4815, CVE-2011-4838, CVE-2012-5372) | Hashes are made non-deterministic by incorporating process start time | Tested in ruby/spec `:security` | Hashes are made non-deterministic by incorporating output from `/dev/random`
+CVE-2011-4838 | JRuby before 1.6.5.1 computes hash values without restricting the ability to trigger hash collisions predictably (cross-reference CVE-2011-4815, CVE-2012-5370, CVE-2012-5372) | Hashes are made non-deterministic by incorporating process start time | Tested in ruby/spec `:security` | Hashes are made non-deterministic by incorporating output from `/dev/random`
```

Comment:
Output from `/dev/random` sounds a little bit weird (it's not a program, it's a pseudo-file).
Maybe simply:
Hashes are made non-deterministic by seeding from `/dev/random` ?